[NFC][Utils] Extract CloneFunctionBodyInto from CloneFunctionInto

artempyanykh opened the pull request:
Stacked PRs:
 * #118630
 * #118629
 * #118628
 * #118627
 * #118625
 * __->__#118624
 * #118623


--- --- ---

### [NFC][Utils] Extract CloneFunctionBodyInto from CloneFunctionInto


Summary:
This and previously extracted `CloneFunction*Into` functions will be used in later diffs.

Test Plan:
ninja check-llvm-unit check-llvm

felipepiovezan commented on llvm/include/llvm/Transforms/Utils/Cloning.h:
```diff
@@ -194,6 +194,15 @@ void CloneFunctionMetadataInto(Function *NewFunc, const Function *OldFunc,
                                ValueMaterializer *Materializer = nullptr);
 
+/// Clone OldFunc's body into NewFunc.
+void CloneFunctionBodyInto(Function *NewFunc, const Function *OldFunc,
```
Can `NewFunc` / `OldFunc` be nullptr? If not, they should be references

artempyanykh replied:
Thanks! Updated them to be references.

felipepiovezan commented on llvm/include/llvm/Transforms/Utils/Cloning.h:
```diff
@@ -361,32 +370,31 @@ void updateProfileCallee(
 /// some noalias scopes needed to be cloned.
-void adaptNoAliasScopes(
-    llvm::Instruction *I, const DenseMap<MDNode *, MDNode *> &ClonedScopes,
-    LLVMContext &Context);
```
are any of these changes part of the PR? It not, they should not be part of the diff.

If you run into this as a result of running `clang-format` in the entire patch, a good way to avoid this is by running `clang-format` on the _diff_ only. For example, (assuming all the changes are  unstaged) `git add` the changes you want to stage, and then `git clang-format --staged`

artempyanykh replied:
Yes, sorry, it was an accidental whole-file format. I backed out those formatting changes.

artempyanykh committed:
[NFC][Utils] Extract CloneFunctionBodyInto from CloneFunctionInto

Summary:
This and previously extracted `CloneFunction*Into` functions will be used in later diffs.

Test Plan:
ninja check-llvm-unit check-llvm

stack-info: PR: https://github.com/llvm/llvm-project/pull/118624, branch: users/artempyanykh/fast-coro-upstream/5
